test/RightsService.Business.UnitTests/Commands/Role/CreateRoleCommandTests.cs

```diff
@@ -1,89 +1,145 @@
-                Users = new List<DbUser> { dbUser }
+                Id = _roleId
             };
 
-            _response = new OperationResultResponse<Guid>
+            _goodResponse = new OperationResultResponse<Guid>
             {
                 Body = _dbRole.Id,
                 Status = OperationResultStatusType.FullSuccess,
                 Errors = new List<string>()
             };
+
+            _autoMocker
+                .Setup<IHttpContextAccessor, IDictionary<object, object>>(x => x.HttpContext.Items)
+                .Returns(_items);
+
+            _autoMocker
+                .Setup<IRoleRepository, Guid>(x => x.Create(_dbRole))
+                .Returns(_roleId);
+
+            _autoMocker
+                .Setup<IDbRoleMapper, DbRole>(x => x.Map(It.IsAny<CreateRoleRequest>(), It.IsAny<Guid>()))
+                .Returns(_dbRole);
+            _autoMocker
+                .Setup<IDbRoleMapper, DbRole>(x => x.Map(It.Is<CreateRoleRequest>(x => x == null), It.IsAny<Guid>()))
+                .Throws(new ArgumentNullException("CreateRoleRequest"));
+
+            _command = new CreateRoleCommand(
+                _autoMocker.GetMock<IHttpContextAccessor>().Object,
+                _autoMocker.GetMock<IRoleRepository>().Object,
+                _autoMocker.GetMock<ICreateRoleRequestValidator>().Object,
+                _autoMocker.GetMock<IDbRoleMapper>().Object,
+                _autoMocker.GetMock<IAccessValidator>().Object);
         }
 
         [SetUp]
         public void SetUp()
         {
-            _mocker.GetMock<ICreateRoleRequestValidator>().Reset();
-            _mocker.GetMock<IAccessValidator>().Reset();
-            _mocker.GetMock<IDbRoleMapper>().Reset();
-            _mocker.GetMock<IRoleRepository>().Reset();
-            _mocker.GetMock<IHttpContextAccessor>().Reset();
-            _mocker.GetMock<IDataProvider>().Reset();
+            _autoMocker
+                .Setup<ICreateRoleRequestValidator, bool>(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(true);
+
+            _autoMocker
+                .Setup<IAccessValidator, bool>(x => x.IsAdmin(null))
+                .Returns(true);
         }
 
-        // TODO
+        [Test]
+        public void ShouldThrowForbiddenExceptionWhenAccessValidatorIsFalse()
+        {
+            _autoMocker
+                .Setup<IAccessValidator, bool>(x => x.IsAdmin(null))
+                .Returns(false);
+
+            Assert.Throws<ForbiddenException>(
+                () => _command.Execute(_newRequest), "Not enough rights.");
+        }
+
+        [Test]
+        public void ShouldThrowValidatorException()
+        {
+            _autoMocker
+                .Setup<ICreateRoleRequestValidator, bool>(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(false);
+
+            Assert.Throws<ValidationException>(
+                () => _command.Execute(_newRequest), "CreateRoleRequest");
```

Comment:
i don't think that write error message is good idea here, because if smbd changed message tests will fall